### PR TITLE
feat: static schema generator

### PR DESCRIPTION
**Pull request by Serpentiel**
## About this change—what it does

adds static schema generator and a way to consume it

## Why this way

to make it easier for us to update and debug services' user config options

### CODE REVIEW

**Comment by Serpentiel** [comment]
acceptance tests: https://github.com/aiven/terraform-provider-aiven/actions/runs/3430064784

File: internal/schemautil/userconfig/apiconvert/fromapi.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+	switch t {
+	// TODO: Uncomment when we use the actual types in the schema.
+	//case "boolean":
+	//	res = false
+	//case "integer":
+	//	res = 0
+	//case "number":
+	//	res = float64(0)
+	//case "string":
+	//	res = ""
```
Comment (byashimov) [inline internal/schemautil/userconfig/apiconvert/fromapi.go:55]: 1. How do we know there is "actual types in the schema"?
2. Can it be uncommented now?

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,7 @@ nav_order: 1
 - Add `aiven_pg_user` import example to docs
 - Extend converter for the service user configuration options `ip_filter` object format
 - Fix `aiven_azure_privatelink_connection_approval` import
+- Add static schema generator for user config options
```
Comment (byashimov) [inline CHANGELOG.md:19]: Please cover all changes in the changelog, including schema changes (at least most noticeable with a ref to this PR).

**Review by byashimov — CHANGES_REQUESTED** [review]
There are still panics there, not critical, but still.

**Comment by Serpentiel** [comment]
> There are still panics there, not critical, but still. Please create a ticket to improve that.

those panics are only reachable via `go generate ./...`, this code isn't being executed in runtime

File: internal/schemautil/userconfig/apiconvert/fromapi.go (hunk 1 above)
Comment (Serpentiel) [reply internal/schemautil/userconfig/apiconvert/fromapi.go:55]: this is planned for v4

File: CHANGELOG.md (hunk 2 above)
Comment (Serpentiel) [reply CHANGELOG.md:19]: there weren't any schema changes in particular, we only added a bunch of new fields—I will update the changelog to cover these

**Review by byashimov — DISMISSED** [review]

**Comment by Serpentiel** [comment]
acceptance tests: https://github.com/aiven/terraform-provider-aiven/actions/runs/3497060099

**Comment by Serpentiel** [comment]
acceptance tests: https://github.com/aiven/terraform-provider-aiven/actions/runs/3497788921

**Review by ivan-savciuc — APPROVED** [review]
LGTM

**Pull request merged**